drivers/SmartThings/matter-appliance/src/matter-oven/init.lua

```diff
@@ -0,0 +1,299 @@
+  return function(driver, device, ib, response)
+    local tn_eps = embedded_cluster_utils.get_endpoints(device, clusters.TemperatureControl.ID,
+      { feature_bitmap = clusters.TemperatureControl.types.Feature.TEMPERATURE_NUMBER })
+    if #tn_eps == 0 then
+      device.log.warn(string.format("Device does not support TEMPERATURE_NUMBER feature"))
+      return
+    end
+    local field = string.format("%s-%d", limit_field, ib.endpoint_id)
+    local val = ib.data.value / 100.0
+
+    -- We clamp the max and min values as per the assumed oven temperature ranges.
+    if val < OVEN_MIN_TEMP_IN_C or val > OVEN_MAX_TEMP_IN_C then
+      if limit_field == "MIN_TEMP" then
+        val = OVEN_MIN_TEMP_IN_C
+      else
+        val = OVEN_MAX_TEMP_IN_C
+      end
+    end
+
+    log.info("Setting " .. field .. " to " .. string.format("%s", val))
+    device:set_field(field, val, { persist = true })
+  end
+end
+
+-- Capability Handlers --
+local function handle_oven_mode(driver, device, cmd)
+  log.info(string.format("handle_oven_mode mode: %s", cmd.args.mode))
+  local ep = component_to_endpoint(device, cmd.component)
+  local supportedOvenModesMap = device:get_field(SUPPORTED_OVEN_MODES_MAP) or {}
+  local supportedOvenModes = supportedOvenModesMap[string.format(ep)] or {}
+  for i, mode in ipairs(supportedOvenModes) do
+    if cmd.args.mode == mode then
+      device:send(clusters.OvenMode.commands.ChangeToMode(device, ep, i - 1))
+      return
+    end
+  end
+  log.warn("handle_oven_mode received unsupported mode: ".." for endpoint: "..ep)
+end
+
+local function handle_temperature_setpoint(driver, device, cmd)
+  local tn_eps = embedded_cluster_utils.get_endpoints(device, clusters.TemperatureControl.ID,
+    { feature_bitmap = clusters.TemperatureControl.types.Feature.TEMPERATURE_NUMBER })
+  if #tn_eps == 0 then
+    device.log.warn(string.format("Device does not support TEMPERATURE_NUMBER feature"))
+    return
+  end
+  device.log.info(string.format("handle_temperature_setpoint: %s", cmd.args.setpoint))
+
+  local value = cmd.args.setpoint
+  local _, temp_setpoint = device:get_latest_state(
+    cmd.component, capabilities.temperatureSetpoint.ID,
+    capabilities.temperatureSetpoint.temperatureSetpoint.NAME,
+    0, { value = 0, unit = "C" }
+  )
+  local ep = component_to_endpoint(device, cmd.component)
+  local min_field = string.format("%s-%d", setpoint_limit_device_field.MIN_TEMP, ep)
+  local max_field = string.format("%s-%d", setpoint_limit_device_field.MAX_TEMP, ep)
+  local min = device:get_field(min_field) or OVEN_MIN_TEMP_IN_C
+  local max = device:get_field(max_field) or OVEN_MAX_TEMP_IN_C
+  local unit = "C"
+  if value <= OVEN_MAX_TEMP_IN_C then
+    device:set_field(CURRENT_CONFIGURED_UNIT, unit, { persist = true })
+    if temp_setpoint.unit == "F" then
+      temp_setpoint.value = utils.f_to_c(temp_setpoint.value)
+      temp_setpoint.unit = unit
+    end
+  else
+    unit = "F"
```

Comment:
This looks like it's creating events using fahrenheit in some cases which I do think we are supposed to be doing. The hub will expect all temperature values to be in celsius and will convert to fahrenheit automatically based on the temperature scale. Am I missing something?

Reply:
You are correct, and I made an update in 4b3418d368c729a37282eb8bc3c49186eb6f3326 to remove the degF handling. Now the capabilities should always use degC. When we receive a capability command, we can simply check if it is over 260 degrees, in which case we interpret it as degF and convert to degC in the driver. Also, devices will always report temp in degC so I removed the check for degF from the attribute handler.

Reply:
Resolve by @nickolas-deboom 's latest commit

Reply:
As discussed, this logic was added back for RPC version 4 or lower.